### DISCUSSION TITLE
404 Error

### DISCUSSION THREAD

**Discussion by barnesb8** [uwrit/leaf#432] [Q&A]
I was able to successfully get the application running locally, but am struggling to get it deployed to IIS. I've followed the directions provided to configure leaf with IIS, but continue to receive a 404 error when I try to browse to the API which I believe causes me to see the "Hmm... No Leaf server was found. Please notify your Leaf administrator." message in the UI. When I check the windows event viewer, I get an "Application '...\leafapi\api\' started successfully." message. However, in the leaf-api log file I am getting a StatusCode 404 error. I do also have fatal errors shown in the log indicating that Leaf is not secured by authorization, but I've compiled the API with the -Debug flag. I've also checked the IIS log, but it did not provide any further details. Anyone have suggestions on other ways to debug this error to determine the issue?

**Comment by ndobb** [comment]
Hi @barnesb8, are you able to post the full contents of the Leaf API log file? That should help us debug.

**Reply by barnesb8** [reply]
[leaf-api-20210707.log](https://github.com/uwrit/leaf/files/6786548/leaf-api-20210707.log)
Attached is the log file that was generated.

**Reply by ndobb** [reply]
Comparing to my local dev `.log` file and one of our deployed servers, I see the first log entry for `"HostingRequestStartingLog"` appears to consistently be `/leafapi/api/config`, which is called when the user logs in, as opposed to `/leafapi`, which I see in your file.

Following https://leafdocs.rit.uw.edu/installation/installation_steps/8b_configure_iis/, can you confirm that:
1. The IIS path to the Leaf 'site' is `"/"`?
    
![image](https://user-images.githubusercontent.com/10424962/124984065-a0072c80-dfed-11eb-899e-e0d83b9c2ed5.png)

2. You've added the rewrite rule to `/leafapi`?

Your configuration in IIS should look like this:
![image](https://user-images.githubusercontent.com/10424962/124984166-bb723780-dfed-11eb-9289-559c672129ab.png)

Basically, `"/"` should return the Leaf client web app, while `"/leafapi/..."` should be rewritten by IIS to `"/api/..."` and return API calls.

**Reply by barnesb8** [reply]
I can confirm that I followed the directions from the link you mentioned. The application path to the leaf site is "/" and the rewrite rule was added to /leafapi. I did have to relocate the rewrite rule to the root leaf site instead of being directly on the leafapi application, which I had previously misconfigured. After making this update and restarting the app pool, I'm still receiving the same log messages, but let me know if I can provide any other information.

**Reply by ndobb** [reply]
That sounds right. In your browser, can you go in the Network tab while starting the app and take a screenshot? I'm looking for some kind of sequence like the example below from which we can determine what calls the app is making to the API and what the point of failure is.

![image](https://user-images.githubusercontent.com/10424962/125143763-8db1ef00-e0d0-11eb-9fb3-91973915125c.png)

**Reply by barnesb8** [reply]
I think this is what you're looking for

![ScreenShot](https://user-images.githubusercontent.com/86612901/125336752-2c7e5b80-e31c-11eb-961e-595458dc4e90.PNG)

**Reply by ndobb** [reply]
Hi @barnesb8, just checking in. Were you able to get this sorted out?

We can optionally do a Zoom debug with you if that would be helpful.
